lib/src/pipeline.rs

```diff
@@ -0,0 +1,60 @@
+use futures::sync::mpsc;
+use futures::{stream,Stream,Sink,Future};
+use std::thread;
+use std::collections::{
+    HashSet,
+    HashMap,
+};
+use std::iter::FromIterator;
+use {
+    Architecture,
+    Program,
+    Function,
+    Region,
+    CallTarget,
+    Rvalue,
+};
+use graph_algos::{
+    VertexListGraphTrait,
+    GraphTrait,
+};
+use std::fmt::Debug;
+//use rayon::prelude::*;
+
+pub fn pipeline<A: Architecture + Debug + 'static >(program: Program, region: Region, config: A::Configuration) -> Box<Stream<Item=Function,Error=()> + Send>
+where A::Configuration: Debug {
+    let (tx,rx) = mpsc::channel::<Function>(10);
+    let name = region.name().clone();
+    thread::spawn(move || {
```

Comment:
You can run them in a thread pool, but that's not mandatory. I only use them to communicate w/ the frontend. I'll probably use rayon to parallelize the disassembly/analysis stage later on.